src/security/decorators.py

```diff
@@ -542,6 +542,9 @@ def wrapper(request, *args, **kwargs):
             deny_access(request)
         elif article_object is not None and (request.user.is_editor(request) or request.user.is_staff):
             return func(request, *args, **kwargs)
+        elif article_object.stage in models.SECTION_EDITOR_STAGES and request.user in article_object.section_editors():
```

Comment:
I think it is safe to remove the stage check and just check `request.user in article_object.section_editors()`